### ISSUE TITLE
Unable to determine Flutter version error for Flutter 3.1.0

### ISSUE THREAD

**Issue by zeshuaro** [subosito/flutter-action#164]
When specifying the `flutter-version` as 3.1.0 in the action, I'm getting the following error on linux:

```
Run subosito/flutter-action@v2.4.0
  with:
    flutter-version: 3.1.0
    cache: true
    cache-key: Linux-flutter-3.1.0-0e67b[2](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:2)27e67e79d303cd7250c9a1efef8a0c959b8a04cf2c06b05828604c4b75
    channel: stable
    cache-path: /opt/hostedtoolcache/flutter
    architecture: x64
  env:
    FLUTTER_VERSION: 3.1.0
    FLUTTER_CHANNEL: stable
Run actions/cache@v2
  with:
    path: /opt/hostedtoolcache/flutter
    key: Linux-flutter-[3](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:3).1.0-0e67b227e67e79d303cd7250c9a1efef8a0c959b8a0[4](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:4)cf2c06b0[5](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:5)828[6](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:6)04c4b[7](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:7)5-stable-3.1.0
  env:
    FLUTTER_VERSION: 3.1.0
    FLUTTER_CHANNEL: stable
Cache not found for input keys: Linux-flutter-3.1.0-0e67b227e67e79d303cd7250c9a1efef[8](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:8)a0c[9](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:9)59b8a04cf2c06b05828604c4b75-stable-3.1.0
Run $GITHUB_ACTION_PATH/setup.sh -c "/opt/hostedtoolcache/flutter" stable 3.1.0 x64
  $GITHUB_ACTION_PATH/setup.sh -c "/opt/hostedtoolcache/flutter" stable 3.1.0 x64
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    FLUTTER_VERSION: 3.1.0
    FLUTTER_CHANNEL: stable
/home/runner/work/_actions/subosito/flutter-action/v2.4.0/setup.sh: line [21](https://github.com/zeshuaro/appainter/runs/6633737268?check_suite_focus=true#step:4:23): [: too many arguments
/home/runner/work/_actions/subosito/flutter-action/v2.4.0/setup.sh: line 21: [: too many arguments
Unable to determine Flutter version for channel: stable version: 3.1.0 architecture: x64
Error: Process completed with exit code 1.
```

**Comment by rk0cc** [comment]
Wait, the latest Flutter release version in stable is `3.0.1` and why applying not released `3.1.0`?

**Comment by zeshuaro** [comment]
Thanks for pointing that out. I think they released [3.1.0](https://github.com/flutter/flutter/releases/tag/3.1.0) on GitHub, but haven't actually published it on flutter.dev. Guess I'll just wait for a bit longer.

**Comment by kuhnroyal** [comment]
You can use 3.1.0 from the beta channel. You have to set the channel argument.
